app/views/produce_items/show.html.erb

```diff
@@ -0,0 +1,10 @@
+<p id="notice"><%= notice %></p>
+
+<h1> <%= @produce_item.name_with_whitespace %> </h1>
```

Comment:
Looking at this here, it feels a lot like we should store this name in a i18n file (like `pt-BR.yml`) and then use the produce_items names as keys. Does that make sense?

Reply:
[is this what you meant?](https://github.com/vasconsaurus/produce_in_season/blob/26735d66e0defcb526958cf28b3f86c9063be9cb/config/locales/pt-BR.yml)

if so, why should we do this?

Reply:
> [is this what you meant?](https://github.com/vasconsaurus/produce_in_season/blob/26735d66e0defcb526958cf28b3f86c9063be9cb/config/locales/pt-BR.yml)

Yes

> if so, why should we do this?

Replacing `_` assumes that that's the only thing we did to [slugify](https://www.semrush.com/blog/what-is-a-url-slug/), while we do know that these version should not contain characters that are not included in `[a-zA-Z0-9-_]` (if there are those, we should either replace or just use the actual name in the database).

Having a dictionary avoids these pitfalls. No logic is needed, we just map keys to values. There is also a nice bonus that it's easy to translate if that's ever the case.

Reply:
huuuum, would we do something like this? 

```
    <% @produce_items.find_each do |produce|  %>
      <tr>
        <td><%= I18n.backend.translations[:'pt-BR'].key produce.name %></td>
...
```

Instead of using readable_name? Though not sure if this is what you thought, I tested it here, but it doesn't seem to work on names that have more than one word. (the first colum in the image)

<img width="421" alt="Screen Shot 2022-06-17 at 18 29 37" src="https://user-images.githubusercontent.com/87862340/174403585-d0a1d6ad-30f8-4a34-bf69-3371a1362868.png">

Reply:
> huuuum, would we do something like this?

Yeap. I'd actually write like `I18n.t(produce.name)`. Even better would be to prefix it, like:

```yml
pt-BR:
  produce_items:
    names:
      abacaxi_havai: 'Abacaxi Havaí',
      ...
```

Then the snippet would become `I18n.t("produce_items.names.#{produce.name}")`.

I think it's fine to put that (`I18n.t...`) inside `ProduceItem#readable_name`, that way we have a nice abstraction :).

BTW, putting the language in the string call (`translations[:'pt-BR']`) goes against the i18n lib purpose. We should rely on the value we set in `config/application.rb` using `config.i18n.default_locale` (it is `:en` by default, but we could change it to pt-BR).

>  I tested it here, but it doesn't seem to work on names that have more than one word. (the first colum in the image)

That's due to using special characters in the yaml keys.

Reply:
Since all our produce is in portuguese, and the seasons are local, I changed our default to be pt-BR. I also translated other english words. 

I think it makes more sense to start with the brazilian version and in the future make the german and english  versions. Also, to that, we'll have to find out the local seasons, and there will be produce we will have to introduce and remove. Right now there are a lot of brazilian produce in our list, that wouldn't make sense for a german user.

Reply:
> Do you think we should abstract both of these into methods?

Nope, we extract them if/when we need to.